plato/draw/Box.py

```diff
@@ -1,38 +1,57 @@
+import collections
 import itertools
 import numpy as np
-from .Lines import Lines
 from .. import math
-from .internal import ShapeDecorator, ShapeAttribute
+from .internal import ShapeDecorator, ShapeAttribute, Shape
+
+_BoxCoordinates = collections.namedtuple(
+    '_BoxCoordinates', ['start_points', 'end_points', 'corners'])
 
 @ShapeDecorator
-class Box(Lines):
+class Box(Shape):
     """A triclinic box frame.
 
-    This primitive draws a triclinic box centered at the origin. It infers
-    dimensionality and can be used to draw a 3D or 2D box, based on whether
-    :code:`Lz` is zero. The box argument must be a tuple or list of length
-    2, 3, or 6, an object with attributes :code:`Lx, Ly, Lz, xy, xz, yz`, or
-    a dictionary with those keys. The values of :code:`Lz, xy, xz, yz` are zero
-    by default.
+    This primitive draws a triclinic box centered at the origin. It is
+    specified in terms of three lattice vector lengths `Lx`, `Ly`,
+    `Lz` and tilt factors, defined using the `hoomd-blue schema
+    <https://hoomd-blue.readthedocs.io/en/stable/box.html>`_.
+
+    Rather than directly initializing via attributes, `Box` objects
+    can also be automatically created from box-type objects using the
+    :py:func:`from_box` method.
 
     Examples::
+
         Lx = Ly = Lz = 10
         xy = xz = yz = 0
-        box_primitive = draw.Box((Lx, Ly, Lz, xy, xz, yz), width, color)
+        box_primitive = draw.Box(Lx=Lx, Ly=Ly, Lz=Lz, width=width, color=color)
+        box_tuple = (Lx, Ly, Lz, xy, xz, yz)
+        box_primitive = draw.Box.from_box(box_tuple)
+
     """
 
-    _ATTRIBUTES = Lines._ATTRIBUTES + list(
+    _ATTRIBUTES = list(
         itertools.starmap(ShapeAttribute, [
-            ('box', np.float32, (1, 1, 1, 0, 0, 0), 1, False,
-             'Box object'),
+            ('Lx', np.float32, 1, 0, False,
+             'Length of first box vector'),
+            ('Ly', np.float32, 1, 0, False,
+             'Length of second box vector'),
+            ('Lz', np.float32, 1, 0, False,
```

Comment:
`Lz = 0` could be a better default value -- if I specified only `Lx` and `Ly`, I would expect a 2D box.